src/agent.md

```diff
@@ -0,0 +1,38 @@
+Module src.agent
+================
+The Jenkins agent relation observer.
+
+Classes
+-------
+
+`AgentRelationData(*args, **kwargs)`
+:   Relation data required for adding the Jenkins agent.
+    
+    Attrs:
+        url: The Jenkins server url.
+        secret: The secret for agent node.
+
+    ### Ancestors (in MRO)
+
+    * builtins.dict
+
+    ### Class variables
+
+    `secret: str`
+    :
+
+    `url: str`
+    :
+
```

Comment:
Is there any changes we can make to make the documentation for these two variable show up here?